wire.md

```diff
@@ -0,0 +1,1008 @@
+
+**UTF-8**: The "UTF-8" encoding scheme outlined in [Chapter 2 Section 5](https://www.unicode.org/versions/Unicode15.0.0/ch02.pdf#G11165) of the Unicode 15.0.0 specification.
+
+## 4. Cryptographic Parameters
+
+### 4.1 BLAKE2b
+The following are the general parameters to be used with BLAKE2b.
+
+- Digest byte length: 32
+- Key byte length (not used): 0
+- Salt (hexadecimal): `5b6b 41ed 9b34 3fe0`
+- Personalization (hexadecimal): `5126 fb2a 3740 0d2a`
+
+### 4.2 Ed25519
+
+Cable uses the Ed25519-SHA-512 variant of [Ed25519][Ed25519]:
+
+- *b* = 256
+- *H* = SHA-512
+- *q* = 2<sup>255</sup> - 19
+- The 255-bit encoding of **F**<sub>2<sup>255</sup>-19</sub> is the usual little-endian encoding of { 0, 1, ..., 2<sup>255</sup> - 20 }
+- ℓ = 2<sup>252</sup> + 27742317777372353535851937790883648493
+- *d* = -121655 / 121666 ∈ **F**<sub>q</sub>
+- *B* is the unique point (*x*,4/5) ∈ *E* for which *x* is positive
+
+## 5. Data Model
+
+### 5.1 Posts
+All of the durable data exchanged over the protocol are composed of posts.
+
+A post MUST have an author (via the required `public_key` field), and MUST
+provide a cryptographic signature (via the required `signature` field) to prove
+that they, in fact, authored it.
+
+When a user "makes a post", they are only writing to some local storage,
+perhaps indexed by the hash of the content for easy querying. Posts are only
+sent to other peers in response to queries about them (e.g. chat messages
+within some time range).
+
+| `post_type` numeric id | common name  | description |
+|------------------------|--------------|-------------|
+| 0                      | `post/text`  | a textual chat message, posted to a channel |
+| 1                      | `post/delete`| the deletion of a previously published post |
+| 2                      | `post/info`  | set or clear informative key/value pairs on a user  |
+| 3                      | `post/topic` | set or clear a channel's topic string |
+| 4                      | `post/join`  | announce membership to a channel |
+| 5                      | `post/leave` | announce cessation of membership to a channel |
+
+#### 5.1.2 Addressing
+Any post in cable can be addressed or referenced by its hash.
+
+The hash for a post is produced by putting a post's verbatim binary content,
+including the post header, through the BLAKE2b function.
+
+#### 5.1.3 Links
+The `links` field in each post's header enables any post to refer to 0 or more
+other posts by means of specifying those posts' hashes.
+
+Referencing a post by its hash provides a **causal proof**: it demonstrates
+that a post must have occurred after all of the other posts referenced. This
+property can be useful for ordering chat messages, since a timestamp alone can
+cause ordering problems if a client's hardware clock is skewed, or a timestamp
+is spoofed.
+
+Implementations are RECOMMENDED to set and utilize links on chat messages
```

Comment:
The idea with recommended implementation is because it adds to the load of work implementors would need to do, which is an accessibility concern, making it possible for more minimal implementations to exist.